crates/polars-arrow/src/compute/take/fixed_size_list.rs

```diff
@@ -214,10 +222,32 @@ pub(super) unsafe fn take_unchecked<O: Index>(
             }
             Some(new_validity.freeze())
         };
-        assert_eq!(count * bytes_per_element, total_bytes);
 
+        assert_eq!(count * bytes_per_element, total_bytes);
         buf.set_len(total_bytes);
 
+        let outer_validity = combine_validities_and(
+            outer_validity.as_ref(),
+            values
+                .validity()
+                // We need at least 1 element as we do `get_bit(i.unwrap_or(0))`
+                .filter(|x| !x.is_empty())
+                .map(|x| {
+                    if indices.has_nulls() {
```

Comment:
Can we dispatch this to a bitmap/boolean gather?

Reply:
Updated to use `take_bitmap_nulls_unchecked` / `take_bitmap_unchecked`

Reply:
Not for this PR, but we should merge these kernels with Polars-compute. Then we can use the branchless boolean gather, which is much, much faster.